presto-iceberg/src/main/java/com/facebook/presto/iceberg/PartitionTransforms.java

```diff
@@ -38,25 +40,61 @@
 import static com.facebook.presto.common.type.Decimals.isShortDecimal;
 import static com.facebook.presto.common.type.Decimals.readBigDecimal;
 import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.TimestampType.TIMESTAMP;
 import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
 import static com.facebook.presto.common.type.VarcharType.VARCHAR;
 import static io.airlift.slice.SliceUtf8.offsetOfCodePoint;
 import static java.lang.Integer.parseInt;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.joda.time.chrono.ISOChronology.getInstanceUTC;
 
 public final class PartitionTransforms
 {
     private static final Pattern BUCKET_PATTERN = Pattern.compile("bucket\\[(\\d+)]");
     private static final Pattern TRUNCATE_PATTERN = Pattern.compile("truncate\\[(\\d+)]");
+    private static final DateTimeField YEAR_UTC = getInstanceUTC().year();
+    private static final DateTimeField MONTH_OF_YEAR_UTC = getInstanceUTC().monthOfYear();
+    private static final DateTimeField DAY_OF_YEAR_UTC = getInstanceUTC().dayOfYear();
+    private static final DateTimeField HOUR_OF_DAY_UTC = getInstanceUTC().hourOfDay();
 
     private PartitionTransforms() {}
 
     public static ColumnTransform getColumnTransform(PartitionField field, Type type)
     {
         String transform = field.transform().toString();
-
-        if (transform.equals("identity")) {
-            return new ColumnTransform(type, Function.identity());
+        switch (transform) {
+            case "identity":
+                return new ColumnTransform(type, Function.identity());
+            case "year":
```

Comment:
Meta comment: I think it would be helpful to reference the spec in the documentation of this class.  I was wondering why the new transformations here all returned `INTEGER` type, and the answer is it's just how it's defined in the spec.